server/src/main/java/org/opensearch/cluster/routing/allocation/allocator/LocalShardsBalancer.java

```diff
@@ -81,6 +84,11 @@ public LocalShardsBalancer(
         this.routingNodes = allocation.routingNodes();
         this.metadata = allocation.metadata();
         avgShardsPerNode = ((float) metadata.getTotalNumberOfShards()) / routingNodes.size();
+        int shardCount = 0;
+        for (ObjectCursor<IndexMetadata> cursor : metadata.indices().values()) {
+            shardCount += cursor.value.getNumberOfShards();
```

Comment:
I do not see a check for primary shard here. Is this based on the assumption that primaries are assigned before replica shards?

Reply:
`getNumberOfShards()` gives number of shards which is number of primaries for the index and corresponds to `index.number_of_shards` index setting.

Reply:
Aah. That makes sense!